docs/src/architecture/notes.md

```diff
@@ -1,73 +1,194 @@
+
+Privately stored notes can only be consumed if the note data is known to the consumer. The note data must be provided to the [transaction kernel](../transactions/transaction-kernel.md). That means, there must be some offchain communication to transmit the note's data from the sender to the recipient.
+
+## Note discovery
+Note discovery describes the process of Miden Clients finding notes they want to consume. It will always be possible to send note data directly to the target account, and if the note was previously recored on-chain, it can be consumed. However, it is also possible to querry the Miden Operator and request newly recorded relevant notes. This is done via Note Tags. Tags are part of the Note's metadata and are represented by a `Felt`. 
+
+The `SyncState` API requires the Miden Client to provide a `note_tag` value which is used as a filter in the response. 
+
+## Note consumption
+As with creation, notes can only be consumed in Miden transactions. If a valid transaction consuming an `InputNote` gets verified by the Miden Node, the note's unique nullifier gets added to the [Nullifier DB](https://0xpolygonmiden.github.io/miden-base/architecture/state.html#nullifier-database) and is therefore consumed. Some notes can be consumed by any account and for some note consumption is restricted.
+
+### Note recipient to restrict note consumption to certain targets
+There are several ways to restrict the set of accounts that can consume a specific note. One way is to specifically define the target account ID as done in the P2ID and P2IDR note scripts. Another way is by using the concept of a `RECIPIENT`. Miden defines a `RECIPIENT` as: `hash(hash(hash(serial_num, [0; 4]), script_hash), input_hash)` represented as `Word`. This concept restricts note consumption to those users who know the pre-image data of `RECIPIENT` - which might be a bigger set than a single account.   
+
+During the [transaction prologue](../transactions/transaction-kernel.md) the users needs to provide all the data to compute the note hash. That means, one can create notes that can only be consumed if the `serial_num` and other data is known. This information can be passed on off-chain by the sender to the recipient. 
+
+For public notes, anyone can compute the note hash and the concept of the `RECIPIENT` doesn't make sense. You can see in the standard [SWAP note script](https://github.com/0xPolygonMiden/miden-base/blob/main/miden-lib/asm/note_scripts/SWAP.masm) how `RECIPIENT` is used. Here, using a single hash, is sufficient to ensure that the swapped asset and its note can only be consumed by the defined target.
```

Comment:
I think that this paragraph should be re-worked. 

It is confusing to me.

Reply:
better?